zipkin-server/src/main/java/zipkin/server/ZipkinHttpCollector.java

```diff
@@ -63,6 +80,47 @@
     return validateAndStoreSpans(encoding, Codec.JSON, body);
   }
 
+  @Singleton
+  @Named
+  public static class RiposteSpanIngestEndpoint extends StandardEndpoint<Void, String> {
+    private final Matcher matcher = Matcher.match("/api/v1/spans", HttpMethod.POST);
+
+    private final ZipkinHttpCollector zipkinHttpCollector;
+
+    @Inject
+    public RiposteSpanIngestEndpoint(ZipkinHttpCollector zipkinHttpCollector) {
+      this.zipkinHttpCollector = zipkinHttpCollector;
+    }
+
+    @Override
+    public CompletableFuture<ResponseInfo<String>> execute(RequestInfo<Void> request, Executor longRunningTaskExecutor,
+                                                           ChannelHandlerContext ctx) {
+      String encoding = request.getHeaders().get("Content-Encoding");
+      byte[] body = request.getRawContentBytes();
+
+      return FutureConverter.toCompletableFuture(
+        zipkinHttpCollector.validateAndStoreSpans(encoding, Codec.JSON, body)
```

Comment:
:) It would be easy to do a real conversion and have the helper methods return CompletableFutures with Riposte ResponseInfos instead of Spring ListenableFutures with ResponseEntity objects, but I didn't think it was worth the effort unless you decide you want to go this route.